From 8be7c34e78bd09fa95ce9972332c8c2bb950ec03 Mon Sep 17 00:00:00 2001
From: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
Date: Tue, 28 Jan 2020 16:14:07 -0800
Subject: [PATCH] ipmi: ipmb_dev_int: add quick fix for raw I2C type
 registration

This commit adds quick fix for cases of run time registration
through sysfs. It should be fixed later by adding parsing code for
ipmb_id instead.

This is a temporary fix. Do not upstream it.

Signed-off-by: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
---
 drivers/char/ipmi/ipmb_dev_int.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/char/ipmi/ipmb_dev_int.c b/drivers/char/ipmi/ipmb_dev_int.c
index 382e4a1df510..c5db37266724 100644
--- a/drivers/char/ipmi/ipmb_dev_int.c
+++ b/drivers/char/ipmi/ipmb_dev_int.c
@@ -328,8 +328,12 @@ static int ipmb_probe(struct i2c_client *client,
 	if (ret)
 		return ret;
 
+#if 1 /* FIXME: Quick fix. Need to add parsing code for ipmb_id instead */
+	ipmb_dev->is_i2c_protocol = true;
+#else
 	ipmb_dev->is_i2c_protocol
 		= device_property_read_bool(&client->dev, "i2c-protocol");
+#endif
 
 	ipmb_dev->client = client;
 	i2c_set_clientdata(client, ipmb_dev);
-- 
2.7.4
